File: kfcore/src/main/java/com/lq/ren/base/ui/reg/RegisterHub.java
package com.lq.ren.base.ui.reg;

import androidx.annotation.NonNull;

import java.util.ArrayList;
import java.util.List;

/**
 * Author lqren on 17/9/14.
 */

public class RegisterHub {

    private final List<Register> registerHub = new ArrayList<>();

    public void add(@NonNull Register register) {
        registerHub.add(register);
    }

    public void clear() {
        for (Register register : registerHub) {
            if (!register.isClear()) {
                register.clear();
            }
        }
        registerHub.clear();
    }
}

File: kfcore/src/main/java/com/lq/ren/base/ui/reg/RegisterHub.kt
package com.lq.ren.base.ui.reg

import java.util.ArrayList

/**
 * Author lqren on 17/9/14.
 */

class RegisterHub {

    private val registerHub = ArrayList<Register>()

    fun add(register: Register) {
        registerHub.add(register)
    }

    fun clear() {
        for (register in registerHub) {
            if (!register.isClear) {
                register.clear()
            }
        }
        registerHub.clear()
    }
}
